### PR TITLE
 removing spotbug issues

### PR DESCRIPTION
**Pull request by hemanttanwar**

### CODE REVIEW

File: eng/spotbugs-aggregate-report/pom.xml [hunk 1]
```diff
@@ -45,6 +45,7 @@
                 <source>..\..\appconfiguration\client\src\main\java</source>
                 <source>..\..\appconfiguration\client\src\samples\java</source>
                 <source>..\..\core\azure-core\src\main\java</source>
+                <source>..\..\core\azure-core\src\samples\java</source>
```
Comment (srnagar) [inline eng/spotbugs-aggregate-report/pom.xml]: If you don't add this, do you still need to have spotbugs exclusions?

Comment (conniey) [reply eng/spotbugs-aggregate-report/pom.xml]: No you don't.

Comment (srnagar) [reply eng/spotbugs-aggregate-report/pom.xml]: Then I would recommend not including snippet code for spotbugs inspections.

Comment (hemanttanwar) [reply eng/spotbugs-aggregate-report/pom.xml]: But we should add this line here so that spotbugs would run for other samples.

**Comment by JonathanGiles** [comment]
This PR was merged a little too eagerly, as discussion still appears to be ongoing (although perhaps more discussion occurred offline). In any case, what I said offline is that I wanted to see a regex that matched all `*JavaDocCodeSnippet` files, and didn't list any Bug patterns, so that it would match against all JavaDocCodeSnippet files and not report any issues with them. The exclusion above is too specific in two different ways, so I would like to see it made more general in a follow-up PR. Also, there should be an XML comment that explains why the suppression is added.

**Pull request merged**